examples/plot_intelex.py

```diff
@@ -0,0 +1,276 @@
+`official documentation
+<https://intel.github.io/scikit-learn-intelex/algorithms.html>`_.
+
+Intel(R) Extension for Scikit-learn accelerates Scikit-learn algorithms by using
+the latest hardware features and optimized caching and threading strategies.
+Find more details in Intel's blog posts on Medium
+(`1 <https://medium.com/intel-analytics-software/save-time-and-money-with-intel-extension-for-scikit-learn-33627425ae4>`_,
+`2 <https://medium.com/intel-analytics-software/accelerate-your-scikit-learn-applications-a06cacf44912>`_).
+In many cases, optimizations translate to hardware from other vendors, albeit
+with smaller performance gains.
+
+For this example, we train two simple
+:class:`sklearn.neighbors.KNeighborsClassifier` instances, one with and one
+without using ``sklearnex``, and compare inference times. Afterward, we upload
+both models to the Hugging Face Model Hub, download them again, and verify the
+optimization survived being saved in the Hub. Hugging Face Hub supports
+``sklearnex``-optimized models, meaning the achieved speedup will translate for
+Inference API users.
+"""
+
+# %%
+# Imports
+# =======
+# First, we import everything required for the rest of this document.
+
+import os
+import pickle
+from pathlib import Path
+from tempfile import mkdtemp
+from time import perf_counter
+from uuid import uuid4
+
+from huggingface_hub import delete_repo, whoami
+from sklearn.datasets import make_classification
+from sklearn.metrics import log_loss
+from sklearn.model_selection import train_test_split
+from sklearnex import patch_sklearn
+
+from skops import card, hub_utils
+
+# %%
+# Data
+# ====
+# Next, we create some generic data. A dataset of 50k rows x 15 columns is big enough
+# to showcase a performance gain from using ``sklearnex``. Generally speaking,
+# larger datasets will benefit more from the ``sklearnex`` optimizations. More
+# details can be found in the official
+# `README <https://github.com/intel/scikit-learn-intelex/blob/master/README.md>`_.
+X, y = make_classification(
+    n_samples=50_000,
+    n_features=15,
+    n_informative=15,
+    n_redundant=0,
+    n_clusters_per_class=1,
+    shuffle=False,
+    random_state=42,
+)
+X_train, X_test, y_train, y_test = train_test_split(
+    X, y, test_size=0.3, random_state=42
+)
+
+# %%
+# Training the stock model
+# ========================
+# Now we can train a stock Scikit-learn
+# :class:`sklearn.neighbors.KNeighborsClassifier`
+
+from sklearn.neighbors import KNeighborsClassifier
+
+clf = KNeighborsClassifier(3)
+clf.fit(X_train, y_train)
+
+# %%
+# Training the optimized model
+# ============================
+# We apply ``patch_sklearn()`` and reimport the model to load the patched
+# version. A message is shown, telling us that Intel(R) Extension for
+# Scikit-learn* has been enabled.
+patch_sklearn()
+from sklearn.neighbors import KNeighborsClassifier
+
+clf_opt = KNeighborsClassifier(3)
+clf_opt.fit(X_train, y_train)
```

Comment:
I think in a script, this is very confusing. It'd be nicer if we explicitly import the class from sklearnex for it to be clear which one it is, and leave a comment that users don't have to change imports, and they can only add the call to patch on top of the file before they run imports.

We can also here show training times and compare them.

Reply:
I changed the imports. Training for kNN is <1s and in many cases `sklearnex` comes with a few percent of overhead. We're working on that, but for now I think it wouldn't help to add it.

Reply:
in that case, it does help to add it, since right now we're only selectively showing data, not the whole truth, and that's not what we wanna do here.

Reply:
I have added a `perf_counter` for the fit stage for completeness. On my machine, both fit times were ~0.05s. The numbers printed, but doing anything else with them (like calculating ratios) won't help if we don't do multiple runs. I hope you're fine with how it is now. Let me know :)